src/arrayMethodSort.js

```diff
@@ -2,10 +2,42 @@
 
 /**
  * Implement method Sort
+ *
+ *
  */
+
+const defaultCompare = function (a, b) {
+  const aNum = String(a);
+  const bNum = String(b);
+
+  if (aNum < bNum) {
+    return -1;
+  }
+
+  if (aNum > bNum) {
+    return 1;
+  }
+
+  if (aNum === bNum) {
+    return 0;
+  }
+};
+
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction = defaultCompare) {
+    for (let i = 0; i < this.length; i++) {
+      for (let j = i + 1; j < this.length; j++) {
+        if (compareFunction(this[i], this[j]) > 0) {
+          const item = this[i];
+
+          this[i] = this[j];
+
+          this[j] = item;
+        }
+      }
+    }
+
+    return this;
```

Comment:
Returning `this` is correct as it allows for method chaining, but you should be aware that the sort method modifies the array in place. If you want to create a new array that is sorted, you would need to clone the array before sorting it.